agent/functional_tests/util/utils_windows.go

```diff
@@ -0,0 +1,778 @@
+}
+
+func GetTaskDefinitionWithOverrides(name string, overrides map[string]string) (string, error) {
+	_, filename, _, _ := runtime.Caller(0)
+	tdDataFromFile, err := ioutil.ReadFile(filepath.Join(path.Dir(filename), "..", "testdata", "taskdefinitions", name, "task-definition.json"))
+	if err != nil {
+		return "", err
+	}
+
+	tdStr := string(tdDataFromFile)
+	for key, value := range overrides {
+		tdStr = strings.Replace(tdStr, key, value, -1)
+	}
+	tdData := []byte(tdStr)
+
+	registerRequest := &ecs.RegisterTaskDefinitionInput{}
+	err = json.Unmarshal(tdData, registerRequest)
+	if err != nil {
+		return "", err
+	}
+
+	tdHash := fmt.Sprintf("%x", md5.Sum(tdData))
+	idempotentFamily := *registerRequest.Family + "-" + tdHash
+
+	existing, err := ECS.DescribeTaskDefinition(&ecs.DescribeTaskDefinitionInput{
+		TaskDefinition: &idempotentFamily,
+	})
+	if err == nil {
+		return fmt.Sprintf("%s:%d", *existing.TaskDefinition.Family, *existing.TaskDefinition.Revision), nil
+	}
+
+	registerRequest.Family = &idempotentFamily
+
+	registered, err := ECS.RegisterTaskDefinition(registerRequest)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s:%d", *registered.TaskDefinition.Family, *registered.TaskDefinition.Revision), nil
+}
+
+type TestAgent struct {
+	Image                string
+	DockerID             string
+	IntrospectionURL     string
+	Version              string
+	ContainerInstanceArn string
+	Cluster              string
+	TestDir              string
+	Logdir               string
+	Options              *AgentOptions
+	Process               *os.Process
+
+	DockerClient *docker.Client
+	t            *testing.T
+}
+
+type AgentOptions struct {
+	ExtraEnvironment map[string]string
+	ContainerLinks   []string
+	PortBindings     map[docker.Port]map[string]string
+}
+
+// RunAgent launches the agent and returns an object which may be used to reference it.
+// It will wait until the agent is correctly registered before returning.
+// 'version' may be a docker image (e.g. amazon/amazon-ecs-agent:v1.0.0) with
+// tag that may be used to run the agent. It defaults to
+// 'amazon/amazon-ecs-agent:make', the version created locally by running
+// 'make'
+func RunAgent(t *testing.T, options *AgentOptions) *TestAgent {
+	agent := &TestAgent{t: t}
+
+	dockerClient, err := docker.NewClientFromEnv()
+	if err != nil {
+		t.Fatal(err)
+	}
+	agent.DockerClient = dockerClient
+
+	tmpdirOverride := os.Getenv("ECS_FTEST_TMP")
+
+	agentTempdir, err := ioutil.TempDir(tmpdirOverride, "ecs_integ_testdata")
+	if err != nil {
+		t.Fatal("Could not create temp dir for test")
+	}
+	logdir := filepath.Join(agentTempdir, "log")
+	datadir := filepath.Join(agentTempdir, "data")
+	os.Setenv("ECS_LOGFILE", logdir + "/log.log")
+	os.Setenv("ECS_DATADIR", datadir)
+	valueName := fmt.Sprintf("test_path_%d", time.Now().UnixNano())
+	fmt.Println("Registry location",valueName)
+  os.Setenv("ZZZ_I_KNOW_SETTING_TEST_VALUES_IN_PRODUCTION_IS_NOT_SUPPORTED", valueName)
```

Comment:
I'll change this `fmt.Println` to `t.Log`.  The tabbing is fixed in a later commit.